server/hazard-service/src/main/java/edu/illinois/ncsa/incore/service/hazard/models/eq/utils/HazardCalc.java

```diff
@@ -334,8 +284,8 @@ public static SeismicHazardResult computeGroundMotionAtSite(Earthquake earthquak
             if (Double.parseDouble(period) == 0.0) {
                 closestHazardPeriod = "0.0";
             }
-
-            Double adjHazardVal = HazardCalc.applyModelEqThresholds(hazardValue, demand, demandUnits, closestHazardPeriod);
+            JSONObject eqThresholds = HazardUtil.toLowerKey(HazardUtil.EARTHQUAKE_THRESHOLDS);
+            Double adjHazardVal = HazardCalc.applyEqThresholds(eqThresholds, hazardValue, demand, demandUnits, closestHazardPeriod);
```

Comment:
If you apply the threshold check here, I think you need to replace "demandUnits" with BaseAttenuation.getUnits(demand) because demandUnits is what the user would like, but at this point, the hazard value from the attenuation returned is in the base units the attenuation model produces.

Reply:
You'll see what I mean if you look at SeismicHazardResult line 33. There it is setting the result unit as the Base attenuation unit.